android/filament-android/src/main/java/com/google/android/filament/MaterialInstance.java

```diff
@@ -232,6 +299,28 @@ public void setParameter(@NonNull String name,
         nSetParameterTexture(getNativeObject(), name, texture.getNativeObject(), sampler.mSampler);
     }
 
+    public static class TextureAndSampler {
```

Comment:
Missing documentation, also the name is awkward. I'd prefer to have separate getTexture/getSampler calls.

Reply:
> Missing documentation

Valid point

> also the name is awkward

Open to suggestions :)

> I'd prefer to have separate getTexture/getSampler calls.

It was done as one call to mirror how it's being set, yet see no reason why not to change that and drop the structure